### PR TITLE
[Translation Support] Apply translations (screens/Customers)

### PR DESCRIPTION
**Pull request by micahtyong**
# In this PR

In this PR, we follow the steps in the ["Applying Our Hook" section](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#a7e5cbd6962e4dbd850b724c3eee7b18) of our [roadmap for translation support](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#cfc52fc0865e4bb2b140e54d395343ba) in order to translate `screens/Customers` for `meepanyar`. 

## Coverage
- [x] `screens/Customers` e.g. (`CustomerMain.tsx`)
- [ ] `screens/Camera`
- [ ] `screens/FinancialSummary`
- [ ] `screens/Home`
- [ ] `screens/Inventory`
- [ ] `screens/Profile`

## How to Review
1. Change your browser device setting to Burmese (or return "my" in `device.ts#detectLanguage`)
2. Run the application and navigate to the screens corresponding to `screens/Customers`. 
3. Ensure nothing is breaking
4. Test in English and ensure wording is still correct

We will translate the rest of `screens/*` in subsequent PRs. 

cc: @julianrkung

### CODE REVIEW

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: I think this is the outdated customer card. The newer one got merged right before bug bash. Can you pull from main?

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx]: fix the merge error

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx]: fix the merge error

File: src/screens/Customers/AddMeterReading.tsx
Comment (julianrkung) [inline src/screens/Customers/AddMeterReading.tsx]: `e.g.` should also be translated right?

File: src/screens/Customers/AddPayment.tsx
Comment (julianrkung) [inline src/screens/Customers/AddPayment.tsx]: NIT: this might look cleaner as a string template so you don't need to do the parentheses concatenations

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: fix merge conflict!

File: src/screens/Customers/CustomerProfile.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerProfile.tsx]: Should these labels / units also have the translation wrapper around them?

File: src/screens/Customers/CustomerProfile.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerProfile.tsx]: Should these labels / units also have the translation wrapper around them?

File: src/screens/Customers/CustomerProfile.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerProfile.tsx]: Should be `Meter Reading` not `current_meter_reading_kwh`

**Review by julianrkung — CHANGES_REQUESTED** [review]
Sorry for the late review. Left some comments. I couldn't run because there were some merge conflicts probably from the PRs that got in right before bug bash. Let me know if you need help resolving them!

Thanks for getting this done :) sorry that the Customer code underwent a pretty large refactor!

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx:206]: I think the `formik.values.selectedMeterType` also needs to be `intl(...)` since it's one of 3 different types of meters (`No Meter`, `Smart Meter`, `Analog Meter`) 

I'm also not sure how `value` works for `input` for a select display. Not sure if it needs to be the same value as one of the `MenuItem` values (which it wouldn't be if we translated it) or if the `value` just determines what is displayed. If it just determines what's displayed, we probably don't need to add `intl` around `formik.values.selectedMeterType`. 

You could test by wrapping `formik.values.selecetedMeterType` in an `intl`, putting some random translation in for it, hardcoding so that it uses the "my" translations and seeing if the record still correctly works (there's probably an easier way to test this core concept so feel free to do some derivative of this).

File: src/components/CustomerCard.tsx
Comment (julianrkung) [inline src/components/CustomerCard.tsx]: I think this is the wrong `CustomerCard`. This file should be deleted

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx]: I think this is wrong, should not say Active [User]

image

File: src/screens/Customers/CustomerProfile.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerProfile.tsx]: Figma & pervious string had this as "Unit Tariff"
image

File: src/screens/Customers/CustomerProfile.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerProfile.tsx:207]: Should also change `per_unit_tariff` to `unit_tariff` here!

File: src/screens/Customers/EditCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomer.tsx]: Put translations around "Edit Customer"

File: src/screens/Customers/EditCustomerInformation.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerInformation.tsx]: Put translation around "Customer Information"

File: src/screens/Customers/CustomerMain.tsx
Comment (julianrkung) [inline src/screens/Customers/CustomerMain.tsx:208]: Github won't let me comment on the line, but make sure to wrap `BaseScreen`'s "Customers" in a translation

File: src/screens/Customers/EditCustomerMeter.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerMeter.tsx:83]: because `formik.values.meterType` is one of `Analog Meter` `Smart Meter` or `No Meter` I think it also needs a translation

edit: actually, check the related note in `AddCustomer`, not sure anymore. Translating might break edit / add customer.

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx]: Should not be `intl(words.meter)` here

File: src/screens/Customers/EditCustomerTariffPlans.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerTariffPlans.tsx:74]: kind of unrelated, but I saw in `TariffPlanCard` there are some untranslated `Ks` and `kWh`, would you mind also translating those in this PR?

**Review by julianrkung — COMMENTED** [review]
Thanks for getting this done! Left some comments and we should be good to merge after

File: src/screens/Customers/AddCustomer.tsx
Comment (micahtyong) [reply src/screens/Customers/AddCustomer.tsx:197]: Gotcha. I used `formik.setFieldValue('selectedMeterType', intl(meterType))` on line 190, and the behavior works as you described!

Comment (micahtyong) [reply src/screens/Customers/AddCustomer.tsx:206]: Resolved!

Comment (julianrkung) [reply src/screens/Customers/AddCustomer.tsx:206]: Wait I don't think that this works. If I change the translated value of "Smart Meter" to "Smart Meterssss" (simulating a burmese translation) in `resource.js` then the selection won't work. We might need to specially map the burmese --> english translations both ways, so we can display it in burmese but send the english variant to airtable so the record is correctly populated. Let's talk about this after team meeting so we can get a solution quickly!

Comment (micahtyong) [reply src/screens/Customers/AddCustomer.tsx:206]: @julianrkung The issue should be resolved now with my most recent commit! 

### Adding a Customer
Now in `AddCustomer.tsx`, we can select the following (pretend we're in Burmese and `Smart Metersss` is Burmese for `Smart Meter`). We know this works because in the following screenshot, the tariffs in `getTariffs()` remained the same. 

Screen Shot 2021-05-01 at 12 13 03 AM

Furthermore, after creating a transaction, we get the following in `meepanyar-node`. 

Screen Shot 2021-05-01 at 12 14 48 AM


### Editing a Customer
Furthermore, `EditCustomer.tsx` seems to function as desired. I can successfully select a new meter while editing a user.
 
Screen Shot 2021-05-01 at 12 15 47 AM

Then, upon moving to the next page, I properly get the tariff plan options from `getTariff()`. 

Screen Shot 2021-05-01 at 12 16 41 AM

Finally, I get the following message in my terminal indicating that the user has been authenticated and successfully updated (as far as I can tell). 

Screen Shot 2021-05-01 at 12 17 15 AM

Let me know how this looks to you!

File: src/lib/i18next/resources.js
Comment (julianrkung) [inline src/lib/i18next/resources.js]: Can you return this back to "Smart Meter" before you merge?

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx]: Can you add a brief comment explaining why these are needed so it's clearer to new devs reading code?

File: src/screens/Customers/EditCustomerMeter.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerMeter.tsx]: `meterType` needs to be mapped here also. So i think it should be ` meterType: currentCustomer?.meterType ? meterTypeMap.get(currentCustomer?.meterType) : ''`

File: src/screens/Customers/EditCustomerMeter.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerMeter.tsx]: the `tariff-plans` route expects a `meterType` prop not `meterTypeKey`, so we need to do `history.push('tariff-plans', { meterNumber, meterType: meterTypeKey, goBack: -2 })`

File: src/screens/Customers/EditCustomerMeter.tsx
Comment (julianrkung) [inline src/screens/Customers/EditCustomerMeter.tsx]: Not technically in scope for this PR, but the `Select` box is cut off and a `marginTop` needs to be added to fix it. Can you change the inline style of `FormControl` to `style={{ width: '100%', marginTop: 8, marginBottom: 8 }}`.

Here's a before & after of what it should look like:

### Before:
image


### After:
image

File: src/screens/Customers/AddCustomer.tsx
Comment (julianrkung) [inline src/screens/Customers/AddCustomer.tsx:124]: Not technically a part of this PR, but all customers made right now are default inactive on the frontend. Could you fix that by adding a `customer.isactive = true`

**Review by julianrkung — APPROVED** [review]
Approved after the changes made to `addCustomer` and `editCustomerMeter`. Make sure to do the edits to `editCustomerMeter` since the latter doesn't work with translations until after the correct `prop` is passed in `handleSubmit` and the correct translation is applied to `meterType` in `formik`

**Pull request merged**
